src/git/Index.cpp

```diff
@@ -424,10 +424,10 @@ Id Index::workdirId(const QString &path, uint32_t *mode) const
 {
   Repository repo(git_index_owner(d->index));
 
-  git_oid id;
-  if (int error = git_repository_hashfile(
-        &id, repo, path.toUtf8(), GIT_OBJECT_BLOB, nullptr))
-    return (error == GIT_EUSER) ? Id::invalidId() : Id();
+  const git_index_entry* entry = git_index_get_bypath(d->index, path.toUtf8(), GIT_INDEX_ENTRY_STAGEMASK);
```

Comment:
Are the changes in this file independent of the other changes (even if they have the same ultimate goal)? If so, please extract them to a separate commit and explain in the commit message how the changes here "work", i.e. how they make things faster (currently I don't understand).

Reply:
They are independent but have the same goal : keep gitahead responding when a large not-commited file is in workdir.
Applying one and not the other will keep gitahead unresponsive in those situation when refreshing workdir, for the duration of the file read and hashing.